TSEconomy/TSEconomy.cs

```diff
@@ -21,21 +22,28 @@ public TSEconomy(Main game) : base(game)
         }
 
         public override void Initialize()
+        {
+            // register hooks
+            ServerApi.Hooks.GameInitialize.Register(this, OnInitialize);
+
+            TShockAPI.Hooks.GeneralHooks.ReloadEvent += (x) => {
+                Configuration.Configuration.Load();
+                x.Player.SendSuccessMessage("[TSEconomy] Reloaded config.");
+
+            };
+
+            // register commands
+            Commands.Commands.RegisterAll();
+        }
+
+        public void OnInitialize(EventArgs args)
         {
             // load our config file
             Configuration.Configuration.Load();
 
             // init db
             DB.InitializeDB(Config.UseMySQL);
 
-            // register commands
```

Comment:
is this change necessary? i may be wrong but GameInitialize occurs when the world loads, i dont see the point of this tbh

Reply:
you may be right -- not totally sure. idk, ideally i'd like for them to keep as little bloat as possible in the codebase, and it doesn't make a ton of sense to split up initialization methods into hooks when it doesn't change functionality for the better. unless i am glossing over some benefit

Reply:
you can check AntiVPN: https://github.com/Maxthegreat99/AntiVPN-TShock/blob/master/Main.cs it sends startup messages after all plugins initialized and is hooking GameInitialize. As for splitting intialization when i wrote this i mostly thought since onInitialize was later on gonna contain some initialization code later on anyway (stuff like checking if world ids are the same etc..) thought it would be good to load it there. now that i think of it since we are making an api it would probably be better loading it when the plugin is actually loading and making sure the plugin loads first as to give a chance for other plugins to do their work when initializing, gonna change it right now